slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -63,10 +63,11 @@ ::grpc::Status SLAMServiceImpl::GetMap(ServerContext *context,
     response->set_mime_type(mime_type);
 
     if (mime_type == "image/jpeg") {
-        std::string jpeg_img = "";
         bool pose_marker_flag = request->include_robot_marker();
         try {
-            jpeg_img = PaintMap(pose_marker_flag);
+            if (!optimizing) {
```

Comment:
The value of `optimizing` could change immediately after this check, so I think `optimizing` needs to be a mutex instead. Then this code would be 

try to lock mutex
if fail, use existing jpeg_image
if succeed, keep mutex locked during call to PaintMap

Reply:
oooh that's smart! I like that.